crates/uv/src/commands/tool/mod.rs

```diff
@@ -16,44 +17,60 @@ pub(crate) mod upgrade;
 
 /// A request to run or install a tool (e.g., `uvx ruff@latest`).
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub(crate) struct ToolRequest<'a> {
-    /// The executable name (e.g., `ruff`), if specified explicitly.
-    pub(crate) executable: Option<&'a str>,
-    /// The target to install or run (e.g., `ruff@latest` or `ruff==0.6.0`).
-    pub(crate) target: Target<'a>,
+pub(crate) enum ToolRequest<'a> {
+    // Running the interpreter directly e.g. `uvx python` or `uvx pypy@3.8`
+    Python {
+        /// The executable name (e.g., `bash`), if the interpreter was given via --from.
+        executable: Option<&'a str>,
+        // The interpreter to install or run (e.g., `python@3.8` or `pypy311`.
+        request: PythonRequest,
+    },
+    // Running a Python package
+    Package {
+        /// The executable name (e.g., `ruff`), if the target was given via --from.
+        executable: Option<&'a str>,
+        /// The target to install or run (e.g., `ruff@latest` or `ruff==0.6.0`).
+        target: Target<'a>,
+    },
 }
 
 impl<'a> ToolRequest<'a> {
     /// Parse a tool request into an executable name and a target.
-    pub(crate) fn parse(command: &'a str, from: Option<&'a str>) -> Self {
-        if let Some(from) = from {
-            let target = Target::parse(from);
-            Self {
-                executable: Some(command),
-                target,
-            }
+    pub(crate) fn parse(command: &'a str, from: Option<&'a str>) -> anyhow::Result<Self> {
+        // If --from is used, the command could be an arbitrary binary in the PATH (e.g. `bash`),
+        // and we don't try to parse it.
+        let (component_to_parse, executable) = match from {
+            Some(from) => (from, Some(command)),
+            None => (command, None),
+        };
+
+        // First try parsing the command as a Python interpreter, like `python`, `python39`, or
+        // `pypy@39`. `pythonw` is also allowed on Windows. This overlaps with how `--python` flag
```

Comment:
Does uvx behave differently with `pythonw`, as uvx is already a terminal command?